src/linux/agent/system-server.js

```diff
@@ -85,12 +86,35 @@ rpc.exports = {
   },
   startActivity: function (pkg, activity) {
     return performOnJavaVM(function () {
-      var intent = packageManager.getLaunchIntentForPackage(pkg);
-      if (intent === null)
+      var appInstalled = false;
+      var apps = packageManager.getInstalledApplications(GET_META_DATA);
+      for (var i = 0; i < apps.size(); i++) {
+        var appInfo = Java.cast(apps.get(i), ApplicationInfo);
+        if (appInfo.packageName.value === pkg) {
+          appInstalled = true;
+          break;
+        }
+      }
+      if (!appInstalled)
         throw new Error("Unable to find application with identifier '" + pkg + "'");
 
-      if (activity !== null)
+      var intent = packageManager.getLaunchIntentForPackage(pkg);
+      if (intent === null && activity === null)
+        throw new Error("Unable to find a front-door activity");
+
+      if (intent === null) {
+        intent = Intent.$new();
+        intent.setAction("android.intent.action.MAIN");
+      }
+
+      if (activity !== null) {
+        var pkgInfo = packageManager.getPackageInfo(pkg, GET_ACTIVITIES);
+        var activities = pkgInfo.activities.value.map((activityInfo) => activityInfo.name.value);
```

Comment:
```suggestion
        var activities = pkgInfo.activities.value.map((activityInfo) => activityInfo.name.value);
```

Should stick to ES5 in the internal agents so they can also be used with Duktape. (Even if we currently use V8 for this particular agent.)